lte/gateway/c/core/oai/tasks/s11/s11_mme_session_manager.cpp

```diff
@@ -15,57 +15,59 @@
  *      contact@openairinterface.org
  */
 
-/*! \file s11_mme_session_manager.c
+/*! \file s11_mme_session_manager.cpp
   \brief
   \author Sebastien ROUX, Lionel Gauthier
   \company Eurecom
   \email: lionel.gauthier@eurecom.fr
 */
 
+#include "lte/gateway/c/core/oai/tasks/s11/s11_mme_session_manager.hpp"
+
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdbool.h>
 #include <stdint.h>
 #include <inttypes.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif
 #include "lte/gateway/c/core/common/assertions.h"
-#include "lte/gateway/c/core/common/dynamic_memory_check.h"
 #include "lte/gateway/c/core/oai/common/conversions.h"
 #include "lte/gateway/c/core/oai/common/log.h"
+#include "lte/gateway/c/core/oai/include/sgw_ie_defs.h"
 #include "lte/gateway/c/core/oai/lib/bstr/bstrlib.h"
-#include "lte/gateway/c/core/oai/lib/hashtable/hashtable.h"
-#include "lte/gateway/c/core/oai/lib/hashtable/hashtable.h"
-#include "lte/gateway/c/core/oai/lib/hashtable/obj_hashtable.h"
-#include "lte/gateway/c/core/oai/lib/itti/intertask_interface.h"
-
 #include "lte/gateway/c/core/oai/lib/gtpv2-c/nwgtpv2c-0.11/include/NwGtpv2c.h"
 #include "lte/gateway/c/core/oai/lib/gtpv2-c/nwgtpv2c-0.11/shared/NwGtpv2cIe.h"
 #include "lte/gateway/c/core/oai/lib/gtpv2-c/nwgtpv2c-0.11/shared/NwGtpv2cMsg.h"
 #include "lte/gateway/c/core/oai/lib/gtpv2-c/nwgtpv2c-0.11/shared/NwGtpv2cMsgParser.h"
-#include "lte/gateway/c/core/oai/include/sgw_ie_defs.h"
+#include "lte/gateway/c/core/oai/lib/hashtable/hashtable.h"
+#include "lte/gateway/c/core/oai/lib/hashtable/obj_hashtable.h"
+#ifdef __cplusplus
+}
+#endif
 
+#include "lte/gateway/c/core/common/dynamic_memory_check.h"
 #include "lte/gateway/c/core/oai/include/mme_app_ue_context.hpp"
-#include "lte/gateway/c/core/oai/tasks/s11/s11_common.h"
-#include "lte/gateway/c/core/oai/tasks/s11/s11_mme_session_manager.h"
-
-#include "lte/gateway/c/core/oai/lib/gtpv2-c/gtpv2c_ie_formatter/shared/gtpv2c_ie_formatter.h"
-#include "lte/gateway/c/core/oai/tasks/s11/s11_ie_formatter.h"
-#include "lte/gateway/c/core/oai/include/s11_messages_types.h"
+#include "lte/gateway/c/core/oai/include/s11_messages_types.hpp"
+#include "lte/gateway/c/core/oai/tasks/s11/s11_common.hpp"
+#include "lte/gateway/c/core/oai/tasks/s11/s11_ie_formatter.hpp"
 
 extern hash_table_ts_t* s11_mme_teid_2_gtv2c_teid_handle;
 
 //------------------------------------------------------------------------------
 status_code_e s11_mme_create_session_request(
     nw_gtpv2c_stack_handle_t* stack_p,
     itti_s11_create_session_request_t* req_p) {
-  nw_gtpv2c_ulp_api_t ulp_req = {0};
+  nw_gtpv2c_ulp_api_t ulp_req = {};
   nw_rc_t rc;
   uint8_t restart_counter = 0;
 
   OAILOG_FUNC_IN(LOG_S11);
 
-  if (!stack_p) return RETURNerror;
+  DevAssert(stack_p);
```

Comment:
Why is this changed?

Reply:
This is changed to keep the code consistent with other functions. In other functions, like `s11_mme_handle_create_session_response`,  `s11_mme_delete_session_request`, `s11_mme_handle_delete_session_response` and `s11_mme_handle_ulp_error_indicatior`, same DevAssert is used and failed with assert if stack_p is null.